### PR TITLE
Add integration test testing upstream TLS client-auth

### PR DESCRIPTION
**Pull request by k-wall**
### Type of change

_Select the type of your PR_

- Enhancement / new feature

### Description

Add integration test testing upstream TLS client-auth.
Builds on top of the work of [callaertanthony](https://github.com/callaertanthony)'s in #844.

### Additional Context

_Why are you making this pull request?_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [x] Write tests
- [x] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [ ] Make sure all Sonar-Lint warnings are addressed or are justifiably ignored.
- [ ] Update documentation
- [ ] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

File: kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/TlsIT.java [hunk 1]
```diff
@@ -162,6 +171,88 @@ void upstreamUsesTlsInsecure(@Tls KafkaCluster cluster) {
...
+                            .endKeyStoreKey()
+                            .endTls()
+                            .endTargetCluster()
+                            .withClusterNetworkAddressConfigProvider(CONFIG_PROVIDER_DEFINITION)
+                            .build());
+
+            try (var tester = kroxyliciousTester(builder); var admin = tester.admin("demo")) {
+                // do some work to ensure connection is opened
+                final var result = admin.describeCluster().clusterId();
+                assertThat(result).succeedsWithin(Duration.ofSeconds(10));
```
Comment (k-wall) [inline kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/TlsIT.java:229]: It'd have been stronger to do something to assert that the client's principal was actually been picked up by Kafka.  However, I don't see an easy way to do this.  ACL rules would work but that brings quite a lot of additional complexity to the test.  I think this  gets over the good enough hurdle.

File: kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/TlsIT.java [hunk 2]
```diff
@@ -162,6 +171,88 @@ void upstreamUsesTlsInsecure(@Tls KafkaCluster cluster) {
...
+            var trustStore = config.get("ssl.truststore.location").toString();
+            var trustPassword = config.get("ssl.truststore.password").toString();
+            assertThat(trustStore).isNotNull();
+            assertThat(trustPassword).isNotNull();
+
+            // Validate a TLS client-auth connection to direct the cluster succeeds/fails as expected.
+            assertSuccessfulDirectClientAuthConnectionWithClientCert(cluster);
+            assertUnsuccessfulClientAuthDirectConnectionWithoutClientCert(cluster);
+
+            var builder = new ConfigurationBuilder()
```
Comment (robobario) [inline kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/TlsIT.java:208]: Maybe the rest of the code should be wrapped up in a method as above like `assertKroxyliciousCanProxyConnectionWithClientCert(...)`. Would it be worth also doing the matching negative test that another Kroxy instance cannot proxy the connection if it's certificates aren't in order?

Comment (k-wall) [reply kroxylicious-integration-tests/src/test/java/io/kroxylicious/proxy/TlsIT.java:208]: Thanks. I tried the suggestion but in the end backed it out. The issue is that whilst the admin connection via the proxy fails, but there's no way for the upstream error why to be communicated back to the downstream.  The test can only assert a failed connection, which doesn't seem to compelling.

 We really need something (exposed through a management layer in Kroxylicious) that reveals the fact that the upstream connection is failing and why.

**Review by robobario — APPROVED** [review]
LGTM

**Pull request merged**